core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -109,6 +110,8 @@
     private final Map<NodeRef<Parameter>, Expression> parameters;
     private String updateType;
     private Optional<UpdateTarget> target = Optional.empty();
+    private Optional<NodeRef<Table>> tableToBeUpdated = Optional.empty();
```

Comment:
from cmt msg

> This makes the check if a table is update target or not independent on current state of the update type.
UpdateType is reset for `EXPLAIN` type of queries and `isUpdateTarget` might not be properly computed.

Does this mean this  fixes a bug?

Reply:
This doesn't fix the actual bug... but rather fixes the issue when we add a validation i.e UPDATE/DELETE is on same pipeline of TableScan.

Reply:
I think that it also fixes an actual bug.
Certain optimizer rules (e.g. `TransformFilteringSemiJoinToInnerJoin`) depend on whether there is a TS being the update target. So, before this change, the `EXPLAIN` plans were different than the actual plans -- that was because in the presence of `EXPLAIN` there was no update target.

Reply:
okay !! I thought the actual bug was not supporting for correlated subquery in `UPDATE` assignment. But yeah it would capture the exact plan for `EXPLAIN` or `EXPLAIN ANALYZE`.

Reply:
I am aware that this fix is a side-effect of the main change, but I think that it should be backed by a test case, and mentioned in the commit message of the relevant commit.